modules/xfeatures2d/include/opencv2/xfeatures2d.hpp

```diff
@@ -948,6 +948,36 @@ CV_EXPORTS void FASTForPointSet( InputArray image, CV_IN_OUT std::vector<KeyPoin
 //! @}
 
 }
+
+#ifdef OPENCV_TRAITS_ENABLE_DEPRECATED
+template<> class DataType<Elliptic_KeyPoint>
+{
+public:
+    typedef Elliptic_KeyPoint      value_type;
+    typedef float                  work_type;
+    typedef float                  channel_type;
+
+    enum {
+        generic_type = 0,
+        depth = DataType<channel_type>::depth,
+        channels = (int)(sizeof(value_type) / sizeof(channel_type)), // 7
```

Comment:
> 7

- Matx23f - 6 floats
- float si - 1 float
- Size_<float> axes - 2 floats
- Keypoint - it just violates assumption about "float" elements - it can't be applied and removed from traits at all.

Reply:
@alalek Any suggestions?